www/src/pages/blog/astro-019.astro

```diff
@@ -0,0 +1,171 @@
+
+          ## `client:only` loading directive
+
+          Sometimes, a UI component can't render on the server. Maybe it's because you've hit a bug in one of your dependencies, or maybe you're just using a library like D3 that can't run without the browser's `window` object. 
+          
+          You can usually wrap browser-only code with your own static server-side fallback UI. However several users wanted a way to render browser-only components without the boilerplate of additional placeholder HTML. Enter `client:only`.
+
+          Astro v0.19 ships with the new `client:only` loading directive to hydrate your component in the browser without server-side rendering. This provides a simple, straightforward fallback for any browser-only components.
+
+          ```html
+          <!-- only renders in the browser, no placeholder HTML -->
+          <MyReactComponent client:only />
+          ```
+
+          In a future release, we plan to add framework auto-detection (needed for multi-framework projects) and explore a more friendly API for generating fallback HTML for browser-only components.
+
+          Thank you Tony Sull (of [Navillus](https://navillus.dev/)) for contributing this new feature. 
+
+          ## `Astro.resolve()`
+
+          Astro 0.19 includes a new [`Astro.resolve()`](https://docs.astro.build/reference/api-reference#astroresolve) helper function to resolve relative file references in your templates. With this new function, you can reference images by relative path on disk and Astro will return the correct URL for the browser.
+          
+          Previously, you had to place files in the `public/` directory and reference them by absolute URL path. Relative paths within the `src/` directory didn't work because they'd be shipped directly to the browser, and different pages would create different URLs:
+
+          ```html
+          <!-- This works the same on every page: -->
+          <img src="/logo.png" />
+          <!-- But this doesn't: -->
+          <img src="../logo.png" />
+          ```
+
+          Starting in Astro 0.19, you can now use the new `Astro.resolve()` helper function to create an absolute URL reference from any relative path:
+          
+          ```jsx
+          <img src={Astro.resolve('../logo.png')} />
+          ```
+
+          If it helps, you can think of `Astro.resolve()` as a simplified alternative to doing `new URL(yourRelativePath, import.meta.url).pathname` in the browser.
+
+          [`Astro.resolve()`](https://docs.astro.build/reference/api-reference#astroresolve) gives you more options and more control over how you structure your project. In the future, this will also unlock our ability to serve optimized images out of the `src/` directory in Astro.
```

Comment:
Would be nice to thank @jonathantneal for starting the original PR for `Astro.resolve`!

Reply:
agreed! done